### PR TITLE
feat(telemetry-processor): normalize raw telemetry events (#81)

### LINKED ISSUE #81: Implement telemetry normalization logic
## Objective
Normalize incoming telemetry data into a consistent internal format.

## Scope
Transform raw events into a standardized internal model.

## Deliverables
- normalization component/service
- mapping logic from raw → normalized event

## Acceptance Criteria
- normalized event structure is consistent
- mapping logic is deterministic
- no data loss during transformation

### PR DESCRIPTION
**Pull request by ME-Massine**
## Summary
Add telemetry normalization logic to transform raw telemetry events into a consistent internal format for downstream processing.

## Related Issue
Closes #81

## Issue Requirements Covered
- Added normalized telemetry event model
- Added normalization service for raw telemetry events
- Added deterministic mapping from raw event to normalized event
- Preserved timestamp and numeric value during transformation
- Added tests proving normalization behavior and null input handling

## Changes
- Added `NormalizedTelemetryEvent`
- Added `TelemetryNormalizationService`
- Updated `TelemetryEventConsumer` to invoke normalization after consuming raw events
- Added unit tests for normalization rules
- Updated consumer test coverage for normalization integration

## Testing
Ran telemetry-processor tests:

```bash
.\mvnw test
```

## Checklist
- [ ] Code builds successfully
- [ ] Tests pass
- [ ] Documentation updated if needed
- [ ] Linked issue is referenced
- [ ] This PR stays within the issue scope

## Summary by Sourcery

Introduce a telemetry normalization layer and integrate it into the telemetry event consumer to produce a consistent internal event format for downstream processing.

New Features:
- Add NormalizedTelemetryEvent model to represent standardized telemetry data.
- Add TelemetryNormalizationService to convert raw telemetry events into normalized events.

Enhancements:
- Update TelemetryEventConsumer to normalize incoming telemetry events and log normalized attributes instead of raw fields.

Tests:
- Add unit tests for TelemetryNormalizationService covering normalization rules, value/timestamp preservation, and null handling.
- Extend TelemetryEventConsumer tests to verify normalization is invoked and not called for null input.

### CODE REVIEW

File: services/telemetry-processor/src/main/java/com/pulsestream/processor/consumer/TelemetryEventConsumer.java [hunk 1]
```diff
@@ -20,12 +28,15 @@ public class TelemetryEventConsumer {
     public void consumeTelemetryEvent(TelemetryEvent telemetryEvent) {
         Assert.notNull(telemetryEvent, "telemetryEvent must not be null");
 
+        NormalizedTelemetryEvent normalizedEvent =
+                normalizationService.normalize(telemetryEvent);
```
[the remark this answers is not in the document]
Comment (ME-Massine) [reply services/telemetry-processor/src/main/java/com/pulsestream/processor/consumer/TelemetryEventConsumer.java:32]: Valid point, but I’m keeping this PR scoped to #81 normalization only. Kafka listener error handling, retry policy, and DLQ behavior should be handled in a dedicated reliability issue so this change does not mix transformation logic with consumer failure strategy.

**Review by MohamedMBG — APPROVED** [review]
LGTM!
